thavalon-server/src/lobby/lobby_impl.rs

```diff
@@ -147,6 +141,36 @@ impl Lobby {
         LobbyResponse::JoinGame(Ok(client_id))
     }
 
+    /// Reconnect a player to an existing game in progress. Helper for add_player.
+    fn reconnect_player(&self, player_id: String, display_name: String) -> LobbyResponse {
```

Comment:
I think for these types, we can make both of them `&str`, instead of `String`. The reason is we only ever use immutable references in the function, and `&String` can be coerced to `&str`, but `&str` cannot be used with `String` types without a new allocation. Admittedly, we don't have any `&str`s in this function, but in the future, we could use them without changing the function signature.